### DISCUSSION TITLE
Community Call —  June 29, 2022

### DISCUSSION THREAD

**Discussion by expede** [ucan-wg/spec#70] [Community Call]
Sign up: https://lu.ma/ucan

# Suggested Agenda

* [ ] UCAN RPC updates

_Let us know in the comments what you'd like to discuss! Questions about using UCAN, use cases in the wild, short demos, etc._

# In Attendance

_Please introduce yourself in the thread below if this is your first time_

# Video

# Chat Log

# Notes

**Comment by expede** [comment]
@dholms
* Made PR for ts-ucan NIST curves (plz be careful!)
* PR into ts-ucan splitting out core-ucan with no dependencies
  * https://github.com/ucan-wg/ts-ucan/pull/82
* Ledger hardware wallet didn't tell you much about what you're signing 👎 
  * Truncates message, even when using with MetaMask
  * Hugo: is the blue box just covering it, rather than hard truncated?
  * https://www.notion.so/blueskyweb/SIWE-Ledger-477b7f46450640d3bfa476cf09031915
* Best way to do SIWE?

@expede 
* WebAuthN/FIDO2
  * Daniel: super excited
  * Benjamin: it works MUCH better today
  * Dmitri: WebAuthN aware of some of the limitations, looking to extend their spec
    * [ ] We should drop in on the WebAuthN dicussions
  * Irakli: PL *may* help get this into browsers
    * Already doing Ed25519

* SIWE/Cacao
  * Irakli: preferred way of handling it is CID in capability
  * Philipp: also can say "implementations MUST be lenient when proofs are in a different format / can get another proof"
  * Irakli: wants to keep parser combinators really clean for good error messages
  * Dmitri: Who's the user? The dev can probably figure it out; end user is going to be much more difficult.
    * Irakli: Means developers
    * Philipp: Could have some signal that it's type A or B, like with DIDs, to give better user messages
    * Irakli: Signalling doesn't help if it could be "anything". We could bake in the namespace, potentially.
    * Sergey: What if we get back to using the same IPLD across UCAN and CACAO? Namespaces, signatures, etc.
    * Irakli: let's try a few concepts, and then have actual data. Likes the idea of unifying, esp if CACAO can do things that UCAN can't or vice versa. Doesn't care about IPLD encoding as much, as long as it can be JWTed.
    * Sergey: Probably the only way to demonstrate this is with to get hands dirty with an end-to-end flow, see the tradeoffs, etc.
    * Daniel: Is CACAO isomorphic to SIWE?
      * Sergey: Yes & no — container for SIWE, working on extending to Solana, signature suites are different, IPLD designators. Must specify the payload & signature (Eth, Solana, etc).
      * Daniel: Does it attenuate?
      * Sergey: No, and this is actually really nice in UCAN :)
      * Benjamin: There's a SIWE/zczp-ld mapping
        * Dmitri: zcap-ld moved away from explicit attenuations, moving to a URI hierarchy (e.g. implicit attenuation in the URI)
          * They're moving towards the claim that REST & RW actions are sufficient to describe *everything*. It's controversial.
          * Mainly by EDV (data vault) community https://www.notion.so/blueskyweb/SIWE-Ledger-477b7f46450640d3bfa476cf09031915
          * Benjamin: learn by cross pollination and concrete experimentation
            * Does UCAN have a prioritized list of use case docs?
* Benjamin: Use cases doc would be useful for figuring out design decisions
  * Doing an early implementation of ActivityPub in UCAN and ZCap-LD, but so far they seem very similar.

* Hugo: HTTP Transport Spec — any movement?
  * Daniel: technically on his backlog
  * Brooke: that would be awesome to have this informed by PL/Bluesky's implementations.

* Hugo: Potential on ramp from pinning API spec. Token could be a UCAN?
  * UCAN doesn't necessarily work perfectly here; need to reuse an invocation

**Reply by dholms** [reply]
Link from Hugo for implementation of HTTP header UCAN transport: https://github.com/web3-storage/ucanto/blob/w3/packages/transport/src/jwt.js
